src/Raven.Server/Documents/Handlers/ShardedDocumentHandler.cs

```diff
@@ -25,29 +35,128 @@ public async Task Get()
 
             HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
 
-            using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            try
             {
+                using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                {
 
-                var id = ids.Single(); // TODO: make it work with multiple shards
+                    var shards = ShardLocator.GetDocumentIdsShards(ids, ShardedContext, context);
+                    foreach (var shard in shards.ShardsToIds)
+                    {
 
-                var shardId = ShardedContext.GetShardId(context, id);
+                        UrlStringBuilder.Clear();
+                        UrlStringBuilder.Append("/docs?");
+                        foreach (var id in shard.Value)
+                        {
+                            UrlStringBuilder.Append($"&id={Uri.EscapeUriString(id)}");
+                        }
 
-                var index = ShardedContext.GetShardIndex(shardId);
+                        var cmd = new ShardedCommand {Method = HttpMethod.Get, Url = UrlStringBuilder.ToString()};
+                        ContextPool.AllocateOperationContext(out TransactionOperationContext ctx);
+                        CmdsContext.Add(ctx);
 
-                var cmd = new ShardedCommand
+                        var task = ShardedContext.RequestExecutors[shard.Key].ExecuteAsync(cmd, ctx);
+                        Task.Add(task);
+                        Cmds.Add(shard.Key, cmd);
+                    }
+
+                    await System.Threading.Tasks.Task.WhenAll(Task);
+
+                    HttpContext.Response.StatusCode = (int)ShardedStatusCodeChooser.GetStatusCode(Cmds.Values);
+
+                    if (HttpContext.Response.IsSuccessStatusCode() == false)
```

Comment:
We need to check if we have errors from any of the other nodes. And if so, we _have_ to give that error to the caller.

Reply:
The first error?

Reply:
The same goes for success status, this is unclear what the behavior should be.

Reply:
Gather all the errors you got.
If there is a single one, throw it.
If multiple, `AggregateException`

All the nodes should return the same status code